### PR TITLE
Initialize uses entered repo

### LINKED ISSUE #486: Settings page "initialize" uses currently stored rather than entered-in-form repo root

### PR DESCRIPTION
**Pull request by isc-etamarch**
Fixes #486

### CODE REVIEW

File: cls/_zpkg/isc/sc/git/Socket.cls [hunk 1]
```diff
@@ -25,6 +25,20 @@ ClassMethod Run()
...
+		set root = %request.Get("root")
+		
+		// Use user input if provided
+		if (root '= "") && ($extract(root,1,3) '= "e.g") {
+			set settings = ##class(SourceControl.Git.Settings).%New()
+			set settings.namespaceTemp = root
+			$$$ThrowOnError(settings.%Save())
+			if ($extract(root, $length(root)) = "\") || ($extract(root, $length(root)) = "/") {
+                set root = $extract(root, 1, $length(root) - 1)
+            }
```
Comment (isc-tleavitt) [inline cls/_zpkg/isc/sc/git/Socket.cls:37]: Spacing here is generally funky

File: csp/gitprojectsettings.csp [hunk 2]
```diff
@@ -658,7 +658,8 @@ function getSocket(urlPostfix) {
 
 function init() {
     disableActionButtons();
-	var ws = getSocket("method=init");
+    var root = document.getElementById("namespaceTemp").value;
```
Comment (isc-tleavitt) [inline csp/gitprojectsettings.csp:661]: Here too

**Review by isc-tleavitt — CHANGES_REQUESTED** [review]
Just a really dumb nitpick here, sorry

**Comment by isc-etamarch** [comment]
Weird, had to delete and re-type the spaces since they looked fine in my editor lol

File: cls/_zpkg/isc/sc/git/Socket.cls [hunk 3]
```diff
@@ -25,6 +25,20 @@ ClassMethod Run()
 	} ElseIf %request.Get("method") = "pull" {
 		Do ##class(SourceControl.Git.API).Pull()
 	} ElseIf %request.Get("method") = "init" {
+		set root = %request.Get("root")
+
+		// Use user input if provided
+		if (root '= "") && ($extract(root,1,3) '= "e.g") {
```
Comment (isc-tleavitt) [inline cls/_zpkg/isc/sc/git/Socket.cls:31]: Wait - why do we need to check for e.g? This is the placeholder, not the value; we wouldn't ever send that?

**Review by isc-tleavitt — COMMENTED** [review]
One more point I missed the first time around.

**Review by isc-tleavitt — APPROVED** [review]

**Pull request merged**